src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/evolutionaryAlgorithm/TerminateByDistancePercBest.java

```diff
@@ -0,0 +1,47 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.evolutionaryAlgorithm;
+
+import java.util.List;
+
+public class TerminateByDistancePercBest implements TerminationCriterion{
+
+    protected final double percBestThreshold;
+    protected final int nrGenerations;
+
+    public TerminateByDistancePercBest( final double percBestThreshold, final int nrGenerations ) {
+        this.percBestThreshold = percBestThreshold;
+        this.nrGenerations = nrGenerations;
+    }
+
+    @Override
+    public boolean readyToTerminate( final int generationNumber, final Generation currentGeneration, final List<Generation> previousGenerations ) {
+        final int preNr = previousGenerations.size();
+        if ( preNr < nrGenerations ) {
+            return false;
+        }
+
+        double bestCostCur = currentGeneration.bestPlan.getWeight();
+        double bestCostPre = previousGenerations.get( preNr-1 ).bestPlan.getWeight();
```

Comment:
I think that `bestCostPrev` is a better name for this variable. or even better: `bestPreviousCost`. When changing this variable name, for symmetry's sake, also change the previous one to `bestCostCurr` or `bestCurrentCost`.

Reply:
Also, when you use the argument `generationNumber` instead of the local variable `preNr` (as I have suggested in my comment above), then here you need to do:
```suggestion
        double bestCostPre = previousGenerations.get( previousGenerations.size()-1 ).bestPlan.getWeight();
```